mobile/components/settings/account/account-balance.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from "react";
+import { KeyInfo } from "@gno/api/gnonativetypes_pb";
+import Text from "@gno/components/text";
+import { useGno } from "@gno/hooks/use-gno";
+
+interface Props {
+  activeAccount: KeyInfo | undefined;
+}
+
+export function AccountBalance({ activeAccount }: Props) {
+  const [address, setAddress] = useState<string | undefined>(undefined);
+  const [balance, setBalance] = useState<string | undefined>(undefined);
+
+  const gno = useGno();
+
+  useEffect(() => {
+    if (!activeAccount) {
+      return;
+    }
+    gno.addressToBech32(activeAccount.address).then((address) => {
+      setAddress(address);
+    });
+    gno.queryAccount(activeAccount.address).then((balance) => {
+      setBalance(JSON.stringify(balance.accountInfo?.coins));
```

Comment:
```suggestion
      setBalance(balance.accountInfo?.coins.reduce(
        (acc, coin) => acc + coin.amount.toString() + coin.denom + " ", ""));
```

Reply:
This imitates the [Coin.String()](https://github.com/gnolang/gno/blob/22fa02a716398bf604f7f994ffce70b87458395c/tm2/pkg/std/coin.go#L59) for a human-friendly display of the coin value(s).